scripts/partition_manager.py

```diff
@@ -55,7 +74,10 @@ def empty_one_of(one_of_list):
                 if len(reqs[k]) == 0:
                     raise PartitionError(empty_one_of(v['one_of']))
             else:
-                resolve_one_of(v, partitions)
+                # Mark as invalid in the next recursion if already invalid or
+                # descending into values of invalid keys.
+                invalid = invalid or k in INVALID_ONE_OF_PROPERTIES
+                resolve_one_of(v, partitions, invalid=invalid)
```

Comment:
Our of curiosity, what concerns you about overwriting the arg?

To be clear, I'm totally fine with what you suggest. I actually had it this way originally, but moved it here to accommodate the 80 char limit without making it less pretty, such as:

```
resolve_one_of(v, partitions, invalid=(
    invalid or k in INVALID_ONE_OF_PROPERTIES))
```

And didn't see an issue with using the arg as that local. Happy to do either way.

Reply:
Mostly readability. It's a bit confusing that it can take a different value, and it _could_ have an effect in the code below. Even if it doesn't, it becomes more complicated to reason about, and it might have an effect later if we refactor.

Reply:
Works for me. Updated, rebased, and squashed into latest commits.